packages/core/src/libraries/hook/index.ts

```diff
@@ -88,51 +127,48 @@ export const createHookLibrary = (queries: Queries) => {
       userIp,
       user: user && pick(user, ...userInfoSelectFields),
       application: application && pick(application, 'id', 'type', 'name', 'description'),
-    } satisfies Omit<HookEventPayload, 'hookId'>;
+    } satisfies InteractionHookEventPayloadWithoutHookId;
 
-    await Promise.all(
-      rows.map(async ({ id, config, signingKey }) => {
-        consoleLog.info(`\tTriggering hook ${id} due to ${hookEvent} event`);
-        const json: HookEventPayload = { hookId: id, ...payload };
-        const logEntry = new LogEntry(`TriggerHook.${hookEvent}`);
+    await sendWebhooks([{ hooks, payload }]);
+  };
 
-        logEntry.append({ hookId: id, hookRequest: { body: json } });
+  /**
+   * Trigger data hooks with the given data mutation context. All context objects will be used to trigger hooks.
+   */
+  const triggerDataHooks = async (hooksManager: DataHookContextManager) => {
+    if (hooksManager.contextArray.length === 0) {
+      return;
+    }
 
-        // Trigger web hook and log response
-        await sendWebhookRequest({
-          hookConfig: config,
-          payload: json,
-          signingKey,
-        })
-          .then(async (response) => {
-            logEntry.append({
-              response: await parseResponse(response),
-            });
-          })
-          .catch(async (error) => {
-            logEntry.append({
-              result: LogResult.Error,
-              response: conditional(
-                error instanceof HTTPError && (await parseResponse(error.response))
-              ),
-              error: conditional(error instanceof Error && String(error)),
-            });
-          });
+    const found = await findAllHooks();
 
-        consoleLog.info(
-          `\tHook ${id} ${logEntry.payload.result === LogResult.Success ? 'succeeded' : 'failed'}`
+    // Filter hooks that match each events
+    const webhookRequests = hooksManager.contextArray
+      .map(({ event, data }) => {
+        const hooks = found.filter(
+          ({ event: hookEvent, events, enabled }) =>
+            enabled && (events.length > 0 ? events.includes(event) : event === hookEvent)
```

Comment:
will it be better to make this part as a method of `ContextManager`?

Reply:
I doubt it. This logic filters/finds the target webhooks data from DB entries using the contextArray.  Shouldn't be a scope of `ContextManager`

Reply:
agreed.